EmscriptenVerBumper/Program.cs

```diff
@@ -0,0 +1,242 @@
+                    "github remote name in the local working directory. Defaults to the same has git_user_name",
+                    v => config.GitRemoteName = v },
+                {"r|reviewers=",
+                    "comma separated list of reviewer usernames",
+                    v => config.Reviewers = v?.Split (",", StringSplitOptions.RemoveEmptyEntries) },
+                { "v|verbose",
+                    "Output information about progress during the run of the tool",
+                    v => config.Verbose = true },
+
+                new ResponseFileSource ()
+            };
+            
+            var remaining = options.Parse (args);
+
+            if (help || args.Length< 1 || remaining.Count > 1) {
+                options.WriteOptionDescriptions (Console.Out);
+                Environment.Exit (0);
+            }
+
+            if (config.GitRemoteName.Length == 0)
+                config.GitRemoteName = config.GitRemoteUserName;
+
+            if (!IsConfigValid (config))
+                Environment.Exit (0);
+
+            return config;
+        }
+
+        static bool IsConfigValid (Configuration config)
+        {
+            if (config.RuntimeWorkingDir.Length == 0) {
+                Console.WriteLine ("Error: --runtime-working-dir required");
+                return false;
+            }
+            if (config.GitRemoteName.Length == 0 || config.GitRemoteUserName.Length == 0) {
+                Console.WriteLine ("Error: Remote name and remote user name are required.");
+                return false;
+            }
+
+            if (!config.DryRun && config.PersonalAccessToken.Length == 0) {
+                Console.WriteLine ($"Error: GitHub personal access token required to create pull requests.");
+                return false;
+            }
+
+            return true;
+        }
+
+        public static async Task<string> GetDockerImage (Configuration config)
+        {
+            string path = config.MasterPath;
+            var versionsBranchHead = await GitHub.GetBranchHead (config.VersionsRepo, config.VersionsBranch, config.Verbose);
+            if (String.IsNullOrEmpty (versionsBranchHead)) {
+                versionsBranchHead = config.VersionsBranch;
+            }
+
+            var masterJsonStream = await GitHub.GetRaw (config.VersionsRepo, versionsBranchHead, config.MasterPath);
+            Console.WriteLine ($"From mono repo {config.VersionsRepo}, branch {config.VersionsBranch} at commit {versionsBranchHead}:");
+
+            var contents = await new StreamReader (masterJsonStream).ReadToEndAsync ();
+            var result = String.Empty;
+            
+            //  Find udpated docker version
+            var match = Regex.Match (contents, config.RegexPattern);
+            if (match.Success) {
+                result = match.Groups[0].ToString();
+            } else {
+                Console.WriteLine($"error: failed to find anything for regex");

```

Comment:
I'm guessing that this should essentially stop execution? You can return `null` here, indicating an error. And the caller can check, and fail.